src/backend/distributed/worker/worker_drop_protocol.c

```diff
@@ -176,6 +165,66 @@ WorkerDropDistributedTable(Oid relationId)
 
 	/* delete the row from pg_dist_partition */
 	DeletePartitionRow(relationId);
+
+	if (!IsObjectAddressOwnedByExtension(&distributedTableObject, NULL))
+	{
+		/*
+		 * If the table is owned by an extension, we cannot drop it, nor should we
+		 * until the user runs DROP EXTENSION. Therefore, we skip dropping the
+		 * table and only delete the metadata.
+		 */
+
+		/*
+		 * Create drop event trigger context to let PG populate objects dropping
+		 * with the upcoming performDeletion command. That objects will be used
+		 * from drop event trigger to unmark objects.
+		 */
+		bool triggerContextCreated = EventTriggerBeginCompleteQuery();
+
+		if (!triggerContextCreated)
+		{
+			ereport(ERROR, (errmsg("Drop trigger context couldn't be created while "
+								   "dropping the distributed table on the worker node")));
+		}
+		PG_TRY();
+		{
+			/*
+			 * We drop the table with cascade since other tables may be referring to it.
+			 */
+			char *relName = get_rel_name(relationId);
+			performDeletion(&distributedTableObject, DROP_CASCADE, 0);
+
+			/*
+			 * Since we drop objects manually with performDeletion, PG doesn't fire
+			 * drop event trigger automatically. Since Citus depends on drop event
+			 * triggers to unmark objects, we are firing it manually
+			 */
+			DropStmt *dropStmt = makeNode(DropStmt);
+			dropStmt->behavior = DROP_CASCADE;
+			dropStmt->removeType = OBJECT_TABLE;
+			dropStmt->missing_ok = false;
+			dropStmt->concurrent = false;
+			dropStmt->objects = list_make1(relName);
+
+			EventTriggerSQLDrop((Node *) dropStmt);
+		}
+		#if PG_VERSION_NUM >= PG_VERSION_13
+		PG_FINALLY();
+		{
+			EventTriggerEndCompleteQuery();
+		}
+		PG_END_TRY();
+		#else
+		PG_CATCH();
+		{
+			EventTriggerEndCompleteQuery();
+			PG_RE_THROW();
+		}
+		PG_END_TRY();
+
+		EventTriggerEndCompleteQuery();
+		#endif
```

Comment:
wondering whether we can simplify this to something like:
```c
       char *relName = get_rel_name(relationId);
        Oid schemaId = get_rel_namespace(relationId);
        char *schemaName = get_namespace_name(schemaId);

        StringInfo dropCommand = makeStringInfo();
        appendStringInfo(dropCommand, "DROP%sTABLE %s CASCADE",
                         IsForeignTable(relationId) ? " FOREIGN " : " ",
                         quote_qualified_identifier(schemaName, relName));

        Node *dropCommandNode = ParseTreeNode(dropCommand->data);

        ProcessUtilityParseTree(dropCommandNode, dropCommand->data, PROCESS_UTILITY_QUERY,
                                NULL, None_Receiver, NULL);
```

Reply:
Done